src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -596,9 +605,12 @@ def reload_module(module):
 
 def get_default_admin_username():
     try:
-        return getpass.getuser()
+        username = getpass.getuser()
     except KeyError:
-        return None
+        username = None
+    if username is None or username.lower() in DISALLOWED_USER_NAMES:
+        username = 'azureuser'
```

Comment:
I would suggest to error out instead of silently using `azureuser` instead. At least a warning should be given.